Tasks/AzureRmWebAppDeployment/AzureRmWebAppDeployment.ts

```diff
@@ -37,6 +39,20 @@ async function run() {
         if (!tl.exist(webDeployPkg)) {
             throw new Error(tl.loc('Invalidwebapppackageorfolderpathprovided', webDeployPkg));
         }
+
+        var isParamFilePresentInPackage = false;
+        var isFolderBasedDeployment = !msDeployUtility.fileExists(webDeployPkg);
+        
+        if (!isFolderBasedDeployment) {
```

Comment:
Some of the checks here are specific to msdeploy. Not sure why they are in the flow during KUDU REST API.